contracts/utils/NoReEntrancy.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { INoReEntrancy } from '../interfaces/INoReEntrancy.sol';
+
+/**
+ * @title NoReEntrancy
+ * @notice This contract provides a mechanism to halt the execution of specific functions
+ * if a pause condition is activated.
+ */
+contract NoReEntrancy is INoReEntrancy {
```

Comment:
Can we make the whole contract similar to the OZ version (names, comments, interfaces etc)?
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol

Makes it easier to audit/monitor as well since we aren't deviating much